src/createClient.js

```diff
@@ -32,7 +32,8 @@ function createClient (options) {
   options.protocolVersion = version.version
   const hideErrors = options.hideErrors || false
 
-  const client = new Client(false, version.minecraftVersion, options.customPackets, hideErrors)
+  const Client = options.customClient ?? DefaultClient
```

Comment:
Why isn't that option enough? Seems like it is to me

Reply:
Isn't it to allow users to provide custom implementation of the Client class? Like you said in https://github.com/PrismarineJS/node-minecraft-protocol/pull/1254#discussion_r1327909079

Reply:
I think I got it. I refactored all it. See latest commits